tsdb/engine/tsm1/engine.go

```diff
@@ -149,27 +155,48 @@ func (e *DevEngine) LoadMetadataIndex(shard *tsdb.Shard, index *tsdb.DatabaseInd
 			measurementFields[measurement] = mf
 		}
 
-		switch typ {
-		case BlockFloat64:
-			if err := mf.CreateFieldIfNotExists(field, influxql.Float, false); err != nil {
-				return err
-			}
-		case BlockInt64:
-			if err := mf.CreateFieldIfNotExists(field, influxql.Integer, false); err != nil {
-				return err
-			}
-		case BlockBool:
-			if err := mf.CreateFieldIfNotExists(field, influxql.Boolean, false); err != nil {
-				return err
+		fieldType, err := tsmFieldTypeToInfluxQLDataType(typ)
+		if err != nil {
+			return err
+		}
+
+		if err := mf.CreateFieldIfNotExists(field, fieldType, false); err != nil {
+			return err
+		}
+
+		_, tags, err := models.ParseKey(seriesKey)
+		if err == nil {
+			return err
+		}
+
+		s := tsdb.NewSeries(seriesKey, tags)
+		s.InitializeShards()
+		index.CreateSeriesIndexIfNotExists(measurement, s)
+	}
+
+	// load the measurement and field metadata from the WAL
+	for measurement, fields := range e.walFields {
+		m := index.CreateMeasurementIndexIfNotExists(measurement)
+
+		mf := measurementFields[measurement]
+		if mf == nil {
+			mf = &tsdb.MeasurementFields{
+				Fields: map[string]*tsdb.Field{},
 			}
-		case BlockString:
-			if err := mf.CreateFieldIfNotExists(field, influxql.String, false); err != nil {
+			measurementFields[measurement] = mf
+		}
+
+		for field, fieldType := range fields {
+			m.SetFieldName(field)
+			if err := mf.CreateFieldIfNotExists(field, fieldType, false); err != nil {
 				return err
 			}
-		default:
-			return fmt.Errorf("unkown block type for: %v. got %v", k, typ)
 		}
+	}
+	e.walFields = nil
 
+	// load the series from the WAL
+	for seriesKey, measurement := range e.walSeries {
```

Comment:
I think the type info might get used via Ben's query refactoring.  Not sure.  You can infer the the type from the value of the points for the key.  You're right that the current TSM code doesn't use it now though.